File: Resources/CourseraAlgorithmicToolbox/week4_divide_and_conquer/6_closest_points/closest.py
#Uses python3
import sys
import math

def minimum_distance(x, y):
    #write your code here
    return 10 ** 18

if __name__ == '__main__':
    input = sys.stdin.read()
    data = list(map(int, input.split()))
    n = data[0]
    x = data[1::2]
    y = data[2::2]
    print("{0:.9f}".format(minimum_distance(x, y)))

File: Resources/CourseraAlgorithmicToolbox/week4_divide_and_conquer/6_closest_points/by_learners/closest.rb
#!/usr/bin/env ruby
# by Andronik Ordian

def min(a, b)
  a < b ? a : b
end

def minimum_distance(x, y)
  # write your code here
  Math.sqrt(10 ** 36)
end

if __FILE__ == $0
  data = STDIN.read.split().map(&:to_i)
  n = data.size - 1
  x = (1..n).step(2).map { |i| data[i] }
  y = (2..n).step(2).map { |i| data[i] }
  puts "#{'%.4f' % minimum_distance(x, y)}"
end